config/import_tree.go

```diff
@@ -27,15 +29,52 @@ type importTree struct {
 // imports.
 type fileLoaderFunc func(path string) (configurable, []string, error)
 
+// Set this to a non-empty value at link time to enable the HCL2 experiment.
+// This is not currently enabled for release builds.
+//
+// For example:
+//    go install -ldflags="-X github.com/hashicorp/terraform/config.enableHCL2Experiment=true" github.com/hashicorp/terraform
```

Comment:
Since this is compile-time rather than a feature flag, would it be easier to mark these with build-constraints, and use something like `go install -tags hcl2`? It may also make it easier as we go to exclude large blocks of code rather than surrounding things with conditionals.

Reply:
For this particular case it worked out simpler to do it this way because that way I don't have to duplicate this entire file to add the conditional branch. The rest of this is designed to be just dead code when this flag isn't set, since there is no codepath leading into the rest.

I agree that this may not be the best strategy for all sorts of feature-flagging scenarios. Honestly, this was just the smallest delta from what I had already written since I didn't expect that this code would live long enough to be worth putting much more effort in for now, though I would like to gather some experience with different approaches here so we can formalize some repeatable patterns that our Makefile and scripts can be aware of to make this easier to use.